citystatmainwindow.cpp

```diff
@@ -0,0 +1,86 @@
+#include "citystatmainwindow.h"
+
+#include "addnewcitydialog.h"
+#include "citystatitemdelegate.h"
+#include "citystatmodel.h"
+#include "ui_citystatmainwindow.h"
+
+CityStatMainWindow::CityStatMainWindow(QWidget *parent) :
+    QMainWindow(parent),
+    ui(new Ui::CityStatMainWindow),
+    m_cityStatisticsModel{new CityStatModel()},
+    m_comboDelegate{new CityStatItemDelegate()},
+    m_addNewCityDialog{new AddNewCityDialog()}
+{
+    ui->setupUi(this);
+
+    ui->cityTableView->setModel(m_cityStatisticsModel);
+    ui->cityTableView->setItemDelegateForColumn(0,m_comboDelegate);
+
+    //table view style
+    ui->cityTableView->setStyleSheet("QHeaderView::section { background-color:#e1e1e1; border: 1px solid #6c6c6c; padding-left: 6px;padding-right: 6px;}");
+    ui->cityTableView->resizeColumnsToContents();
+
+    //connections
+    connect(m_addNewCityDialog, SIGNAL(addNewItem(QStringList)), this, SLOT(onNewItemAdded(QStringList)));
```

Comment:
Use modern connects like connect(this, &CityStatMainWindow::initiateDataSave, m_cityStatisticsModel, &CityStatModel::saveDataInitiated);